src/System.IO.Pipelines/src/System/IO/Pipelines/PipeCompletion.cs

```diff
@@ -19,6 +19,7 @@ internal struct PipeCompletion
         private bool _isCompleted;
         private ExceptionDispatchInfo _exceptionInfo;
 
+        private PipeCompletionCallback _firstCallback;
```

Comment:
Use a single object field? I'm going to look at reducing the size of the pipe, it's one of the bigger objects we allocate today...

Reply:
Callback is action and a state and `PipeCompletionCallback ` is a struct.